django/db/backends/postgresql_psycopg2/client.py

```diff
@@ -7,13 +11,40 @@ class DatabaseClient(BaseDatabaseClient):
     executable_name = 'psql'
 
     def runshell(self):
+        def escape_pgpass(txt):
+            return txt.replace('\\', '\\\\').replace(':', '\\:')
+
         settings_dict = self.connection.settings_dict
         args = [self.executable_name]
-        if settings_dict['USER']:
-            args += ["-U", settings_dict['USER']]
-        if settings_dict['HOST']:
-            args.extend(["-h", settings_dict['HOST']])
-        if settings_dict['PORT']:
-            args.extend(["-p", str(settings_dict['PORT'])])
-        args += [settings_dict['NAME']]
+
+        host = settings_dict.get('HOST', '')
+        port = settings_dict.get('PORT', '')
+        name = settings_dict.get('NAME', '')
+        user = settings_dict.get('USER', '')
+        passwd = settings_dict.get('PASSWORD', '')
+
+        if user:
+            args += ["-U", user]
+        if host:
+            args += ["-h", host]
+        if port:
+            args += ["-p", str(port)]
+        args += [name]
+
+        if passwd:
+            # Create temporary .pgpass file
+            # It's deleted automagically on sys.exit
+            temp_pgpass = tempfile.NamedTemporaryFile(mode='w+')
```

Comment:
Most likely this will not work on Windows because files created with `NamedTemporaryFile` cannot be reopened on Windows (which defeats the whole purpose of naming them in the first place -- I have no idea why `NamedTemporaryFile` even exists on Windows).

I'm not saying this is blocking the merge because I don't think we have that many users of PostgreSQL on Windows, but I thought I'd bring it up in case someone wants to check.